composer/datasets/streaming/download.py

```diff
@@ -27,23 +27,13 @@ def download_from_s3(remote: str, local: str, timeout: float) -> None:
         local (str): Local path (local filesystem).
         timeout (float): How long to wait for shard to download before raising an exception.
     """
-    try:
-        import boto3
-        from botocore.config import Config
-        from botocore.exceptions import ClientError
-    except ImportError as e:
-        raise MissingConditionalImportError(extra_deps_group='streaming', conda_package='boto3') from e
-
     obj = urllib.parse.urlparse(remote)
     if obj.scheme != 's3':
         raise ValueError(f"Expected obj.scheme to be 's3', got {obj.scheme} for remote={remote}")
-
-    config = Config(read_timeout=timeout)
-    s3 = boto3.client('s3', config=config)
-    try:
-        s3.download_file(obj.netloc, obj.path.lstrip('/'), local)
-    except ClientError as e:
-        _ensure_not_found_errors_are_wrapped(remote, e)
+    client_config = {'read_timeout': timeout}
+    bucket, key = obj.netloc, obj.path.lstrip('/')
+    object_store = S3ObjectStore(bucket=bucket, client_config=client_config)
+    object_store.download_object(key, local)
 
 
 def download_from_sftp(remote: str, local: str) -> None:
```

Comment:
Curious if we still need the `download_from_sftp` / `download_from_s3` / `download_from_local` helper functions, or whether it would be possible to use `composer.utils.file_helpers.get_file` instead? Thinking that we could make the `get_file` helper function smarter to handle `s3://` and `sftp://` URIs (and auto-construct the appropriate object store if a URI begins with one of these schemes). Realize that this is slightly beyond the scope of this PR, so OK to defer to another one, but would like `get_file` to be smart and be able to handle `s3://` and `sftp://` URIs too (this will help with checkpointing).

Reply:
My understanding is the `get_file` utility function is being refactored by @eracah -- happy to take a look but I don't want to make a conflict unnecessarily, can we defer this to the next PR?